Add option to configure default platforms

ReToCode opened the pull request:
Adds an option to configure default platforms either in `.ko.yaml` or using the env variable `KO_DEFAULTPLATFORMS`.

Fixes #521

ReToCode commented:
@imjasonh something like this? Unfortunately there is no easy way to test this properly without either
* exposing internals of `gobuildOpener` to test this from `resolver_test.go`
* or extending the `build.Interface` interface similar to `QualifyImport` to test it in `resolver_test.go`

WDYT?

imjasonh commented:
This looks great! This is honestly even less code than I thought it would be. 👍 

The test seems fine. Maybe worth adding another `.ko.yaml` that has `defaultPlatforms: foo,bar` just to make sure it handles splitting on `,` in the config as we expect.

ReToCode commented:
I added another test case but not with splitting `,` as viper expects a yaml array to be parsed as a string slice. I also changed the docs accordingly.

imjasonh reviewed (dismissed):
Nice, that test covers exactly what I was hoping to find out!

imjasonh commented on pkg/commands/options/testdata/multiple-platforms/.ko.yaml:
```diff
@@ -0,0 +1,4 @@
+defaultBaseImage: alpine
+defaultPlatforms: 
+- linux/arm64
+- linux/amd64
```
```suggestion
- linux/amd64

```

ReToCode commented:
Seems like there was already an empty line in that file which is not really visible in GitHub.

imjasonh commented:
Thanks again for this contribution, and for fighting through the lint checks for it! 😅
